src/idl_gen_cpp.cpp

```diff
@@ -2522,6 +2524,16 @@ class CppGenerator : public BaseGenerator {
                 ? ".type"
                 : (field.value.type.element == BASE_TYPE_UNION ? ".value" : "");
         code += "{ _o->" + name + ".resize(_e->size()); ";
+
+        if (!field.value.type.enum_def && IsOneByte(field.value.type.element)) {
+          // For vectors of bytes, std::copy is used to improve performance.
+          // This doesn't work for:
+          //  - enum types because they have to be explicitly static_cast.
+          //  - multiple-byte types due to endianness.
+          code += "std::copy(std::begin(*_e), std::end(*_e), std::begin(_o->" +
```

Comment:
The other thing you can look at is the `std::vector::insert` and `std::vector::assign` instead.

Reply:
These were significantly less performant (at least with the standard library implementation I am using).